src/jit/codegencommon.cpp

```diff
@@ -10583,7 +10583,8 @@ GenTreePtr CodeGen::genMakeConst(const void* cnsAddr, var_types cnsType, GenTree
 //             funclet frames: this will be FuncletInfo.fiSpDelta.
 void CodeGen::genPreserveCalleeSavedFltRegs(unsigned lclFrameSize)
 {
-    regMaskTP regMask = compiler->compCalleeFPRegsSavedMask;
+    genVzeroupperIfNeeded(false);
```

Comment:
Consider a managed method is using scalar floating point operations but no SIMD (128 or 256-bit) and running on AVX hardware.  JIT would produce *scalar* floating point AVX instructions for the method.  In the method prolog we are generating vzeroupper because the logic in lowerxarch would set a flag on emitter to indicate, method uses  AVX instructions (but no 256-bit).

One reason i could think of is reverse PInvoking such a managed method.  The native code could be using legacy SSE (scalar or 128-bit) and calling to managed method could result in legacy SSE to *scalar* AVX transition.  Can you please confirm that will incur transition penalty?  It is not clear to me what Intel manual says in this regard.

---
In reply to: [95486702](https://github.com/dotnet/coreclr/pull/8588#discussion_r95486702) [](ancestors = 95486702)